flake.nix

```diff
@@ -0,0 +1,150 @@
+{
+  description = "Nix package for centrifuge-chain";
+
+  inputs = {
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-21.11;
+    flake-utils = {
+        url = github:numtide/flake-utils;
+        inputs.nixpkgs.follows = "nixpkgs";
+    };
+    gitignore = {
+      url = github:hercules-ci/gitignore.nix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    fenix = {
+      url = github:nix-community/fenix;
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs :
+    inputs.flake-utils.lib.eachDefaultSystem (system:
+        let
+          name = "centrifuge-chain";
+          # This is the program version.
+          version = "0.10.18";
```

Comment:
Is it possible to extract this from the source instead of needing to update our version in another place? (I need to learn more nix, obviously)

Reply:
As far as I know (and tried with my limited knowledge) is that, generally, arguments cannot be passed to flakes. There are a few discussions open around this topic: https://github.com/NixOS/nix/issues/5663

In any case, I would love to not have neither the version nor the nightly defined here. Maybe @mpontus have an idea around this?